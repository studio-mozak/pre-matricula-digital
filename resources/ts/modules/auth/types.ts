export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
