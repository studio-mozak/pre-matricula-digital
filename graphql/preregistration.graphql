extend type Query @guard(with: ["prematricula"]) {

  """
  Lista as pré-matrículas.
  """
  preregistrations(
    """
    Retorna apenas as pré-matrículas onde o **protocolo** ou o **nome do(a) aluno(a)** contenha `search`.
    """
    search: String @slug @search(columns: ["protocol", "student.slug"])
    process: ID @eq(key: "process_id")
    processes: [ID!] @in(key: "process_id")
    school: ID @eq(key: "school_id")
    schools: [ID!] @in(key: "school_id")
    grade: ID @eq(key: "grade_id")
    grades: [ID] @in(key: "grade_id")
    period: ID @eq(key: "period_id")
    type: PreRegistrationType @eq(key: "preregistration_type_id")
    status: PreRegistrationStatus @eq(key: "status")
    sort: PreRegistrationSort @scope(name: "sort")
    year: Int @search(columns: ["process.school_year_id"])
  ): [PreRegistration!]! @paginate

}

extend type Query @guard(with: ["web", "sanctum", "prematricula"]) {

  """
  Retorna a pré-matrícula do protocolo.
  """
  preregistrationByProtocol(
    protocol: String @eq
    withVacancy: Boolean
  ): PreRegistration @first

  """
  Retorna a inscrição usando o código de verificação.
  """
  preregistrationByCode(
    code: String @eq
  ): PreRegistration @first

  getStudentMatches(
    stage: ID!
    name: String @slug
    dateOfBirth: Date @rename(attribute: "date_of_birth")
    cpf: String
    rg: String
    birthCertificate: String @rename(attribute: "birth_certificate")
  ): [StudentMatch!]!

  summonNextInLine(
    process: ID! @eq(key: "process_id")
    school: ID! @eq(key: "school_id")
    grade: ID! @eq(key: "grade_id")
    period: ID! @eq(key: "period_id")
  ): PreRegistration @first(scopes: ["nextInLine", "waiting"])

  matchByStudentCpf(
    cpf: String! @eq
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentRg(
    rg: String! @eq
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentBirthCertificate(
    birthCertificate: String! @eq(key: "birth_certificate")
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentNameAndDateOfBirth(
    name: String! @slug @eq(key: "slug")
    dateOfBirth: Date! @eq(key: "date_of_birth")
    year: Int! @eq
  ): RegistrationMatch @first

  findPreRegistrationsByStudentCpf(
    cpf: String!
  ): [PreRegistration!]!

  findPreRegistrationsByStudentRg(
    rg: String!
  ): [PreRegistration!]!

  findPreRegistrationsByStudentBirthCertificate(
    birthCertificate: String! @rename(attribute: "birth_certificate")
  ): [PreRegistration!]!

  findPreRegistrationsByStudentNameAndDateOfBirth(
    name: String! @slug
    dateOfBirth: Date! @rename(attribute: "date_of_birth")
  ): [PreRegistration!]!

  findPreRegistrationsByInfo(
    cpf: String @rename(attribute: "responsible_cpf")
    email: String @rename(attribute: "responsible_email")
    protocol: String
  ): [PreRegistration!]!

  getPreRegistrationLinkedByEmail(
    email: String! @eq
  ): [PreRegistrationLinkedByEmail!]! @all

  searchLegacyStudentsByName(
    name: String!
    first: Int
  ): [LegacyStudent!]!
}

enum PreRegistrationType {
  """
  Rematrícula
  """
  REGISTRATION_RENEWAL @enum(value: 1)
  """
  Matrícula
  """
  REGISTRATION @enum(value: 2)
  """
  Lista de espera
  """
  WAITING_LIST @enum(value: 3)
}

enum PreRegistrationStatus {
  """
  Pré-matrícula aguardando (in)deferimento.
  """
  WAITING @enum(value: 1)

  """
  Pré-matrícula deferida.
  """
  ACCEPTED @enum(value: 2)

  """
  Pré-matrícula indeferida.
  """
  REJECTED @enum(value: 3)

  """
  Pré-matrícula convocada.
  """
  SUMMONED @enum(value: 4)

  """
  Pré-matrícula em confirmação.
  """
  IN_CONFIRMATION @enum(value: 5)
}

enum RelationType {
  """
  Mãe
  """
  MOTHER @enum(value: 1)

  """
  Pai
  """
  FATHER @enum(value: 2)

  """
  Responsável legal
  """
  GUARDIAN @enum(value: 3)

  """
  A própria pessoa
  """
  SELF @enum(value: 4)
}

enum PreRegistrationSort {
  DATE @enum(value: 1)
  POSITION @enum(value: 2)
  SCHOOL @enum(value: 3)
  NAME @enum(value: 4)
  DATE_OF_BIRTH @enum(value: 5)
}

type RegistrationMatch {
  id: ID!
  year: Int!
  school: ID!
  grade: ID!
  period: ID!
  initials: String!
  dateOfBirth: Date! @rename(attribute: "date_of_birth")
}

enum StudentMatchType {
  PRE_REGISTRATION_ALREADY_EXISTS @enum(value: 1)
  REGISTRATION_CURRENT_YEAR @enum(value: 2)
  REGISTRATION_LAST_YEAR @enum(value: 3)
  STUDENT_ALREADY_EXISTS @enum(value: 4)
}

type StudentMatch {
  id: ID!
  initials: String!
  dateOfBirth: Date! @rename(attribute: "date_of_birth")
  type: ProcessStageRestriction!
  process: Process
  registration: StudentMatchRegistration
}

type StudentMatchRegistration {
  year: Int!
  school: School!
  grade: Grade!
  period: Period
}

type PreRegistration {
  id: ID!
  type: PreRegistrationType! @rename(attribute: "preregistration_type_id")
  position: Int!
  parent: PreRegistration @belongsTo
  waiting: PreRegistration @belongsTo
  others: [PreRegistration]
  protocol: String!
  code: String!
  date: DateTime! @rename(attribute: "created_at")
  process: Process @belongsTo
  period: Period @belongsTo
  school: School @belongsTo
  grade: Grade @belongsTo
  schoolYear: SchoolYear @belongsTo
  classroom: Classroom @belongsTo
  inClassroom: Classroom @belongsTo
  student: Person @belongsTo
  responsible: Person @belongsTo
  relationType: RelationType! @rename(attribute: "relation_type_id")
  status: PreRegistrationStatus!
  observation: String
  fields: [PreRegistrationField!]! @hasMany
  stage: ProcessStage @belongsTo
  external: ExternalPerson @hasOne
  linked: [PreRegistrationLinkedByEmail!]! @hasMany
}

type PreRegistrationField {
  id: ID!
  preregistration: PreRegistration! @belongsTo
  field: Field! @belongsTo
  value: String
}

type PreRegistrationLinkedByEmail {
  id: ID!
  preregistration: PreRegistration! @belongsTo
  email: String!
}

extend type Mutation @guard(with: ["web", "sanctum", "prematricula"]) {

  newPreRegistration(
    input: PreRegistrationInput! @spread
  ): [PreRegistration!]!

  sendProtocolsByEmail(
    preregistrations: [ID!]!
    email: String!
  ): Boolean!

  keepOnTheWaitingList(
    id: ID!
    grade: ID
  ): Boolean!

  updateStudentInExternalSystem(
    preregistration: ID!
    cpf: Boolean
    rg: Boolean
    birthCertificate: Boolean
    name: Boolean
    dateOfBirth: Boolean
    gender: Boolean
    phone: Boolean
    mobile: Boolean
    address: Boolean
  ): Boolean!

  savePreRegistrationLinkedByEmail(
    preregistration: ID! @rename(attribute: "preregistration_id")
    email: String!
  ): PreRegistrationLinkedByEmail!

  deletePreRegistrationLinkedByEmail(
    id: ID! @whereKey
  ): PreRegistrationLinkedByEmail! @delete

}

extend type Mutation @guard {
  acceptPreRegistrations(
    ids: [ID!]!
    classroom: ID!
  ): [PreRegistration!]!

  rejectPreRegistrations(
    ids: [ID!]!
    justification: String
  ): [PreRegistration!]!

  summonPreRegistrations(
    ids: [ID!]!
    justification: String
  ): [PreRegistration!]!

  returnToWaitPreRegistrations(
    ids: [ID!]!
  ): [PreRegistration!]!

  updateStudent(
    protocol: String!
    fields: [PreRegistrationFieldInput!]!
  ): PreRegistration

  updateResponsible(
    protocol: String!
    fields: [PreRegistrationFieldInput!]!
  ): PreRegistration

  updatePreRegistration(
    protocol: String!
    grade: ID!
    school: ID!
    period: ID!
  ): PreRegistration

  updateAddress(
    protocol: String!
    address: AddressInput
  ): PreRegistration
}

input PreRegistrationFieldInput {
  field: ID!
  value: String
}

input AddressInput {
  postalCode: String! @rename(attribute: "postal_code")
  address: String!
  number: String!
  complement: String
  neighborhood: String!
  city: String
  stateAbbreviation: String
  cityIbgeCode: Int
  lat: Float! @rename(attribute: "latitude")
  lng: Float! @rename(attribute: "longitude")
  manualChangeLocation: Boolean! @rename(attribute: "manual_change_location")
}

input PreRegistrationWaitingListInput {
  period: ID! @rename(attribute: "period_id")
  school: ID @rename(attribute: "school_id")
}

input PreRegistrationInput {
  type: PreRegistrationType!  @rename(attribute: "preregistration_type_id")
  process: ID! @rename(attribute: "process_id")
  stage: ID! @rename(attribute: "process_stage_id")
  grade: ID! @rename(attribute: "grade_id")
  period: ID! @rename(attribute: "period_id")
  school: ID @rename(attribute: "school_id")
  optionalSchool: ID
  optionalPeriod: ID
  address: AddressInput!
  optionalAddress: AddressInput
  relationType: RelationType! @rename(attribute: "relation_type_id")
  responsible: [PreRegistrationFieldInput!]!
  student: [PreRegistrationFieldInput!]!
  externalPerson: ID @rename(attribute: "external_person_id")
  waitingList: [PreRegistrationWaitingListInput!]
}

type LegacyStudent {
  id: ID!
  name: String!
  cpf: String
  dateOfBirth: String @rename(attribute: "date_of_birth")
  responsibleName: String @rename(attribute: "responsible_name")
  responsibleCpf: String @rename(attribute: "responsible_cpf")
  responsibleDateOfBirth: String @rename(attribute: "responsible_date_of_birth")
  responsiblePhone: String @rename(attribute: "responsible_phone")
}
