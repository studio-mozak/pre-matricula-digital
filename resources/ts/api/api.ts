import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
