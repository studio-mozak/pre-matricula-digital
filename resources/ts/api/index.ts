export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
