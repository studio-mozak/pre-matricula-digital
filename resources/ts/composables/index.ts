export * from './useLoader';
export * from './useLoaderAndShowErrors';
export * from './usePortal';
export * from './useModal';
export * from './useStudentProcessAndSuggestGrades';
