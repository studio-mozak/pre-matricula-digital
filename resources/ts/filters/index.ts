import { Address, GoogleMapsMarker } from '@/types';
import {
  getFormattedDate,
  getFormattedDateTime,
  getFormattedTime,
} from '@/datetime';
import {
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
} from '@/util';

export interface Filters {
  formatDateTime: (date: string) => string;
  formatDate: (date: string) => string;
  formatTime: (date: string) => string;
  joinAddress: (address: Address) => string;
  joinAddressLine1: (address: Address) => string;
  joinAddressLine2: (address: Address) => string;
  joinAddressLine3: (address: Address) => string;
  markerAddress: () => GoogleMapsMarker;
}

const filters: Filters = {
  formatDateTime(date: string) {
    return getFormattedDateTime(date);
  },
  formatDate(date: string) {
    return getFormattedDate(date);
  },
  formatTime(date: string) {
    return getFormattedTime(date);
  },
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
};

export default filters;
