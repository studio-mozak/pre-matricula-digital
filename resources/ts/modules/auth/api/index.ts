/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
