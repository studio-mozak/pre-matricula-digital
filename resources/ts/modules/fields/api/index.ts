/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
