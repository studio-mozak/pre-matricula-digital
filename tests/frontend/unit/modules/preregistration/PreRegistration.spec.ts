/// <reference types="@types/google.maps" />
import '@/plugin/sortby';
import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import { useRoute, useRouter } from 'vue-router';
import PreRegistration from '@/modules/preregistration/components/PreRegistration.vue';
import { PreRegistrationStage } from '@/modules/preregistration/types';
import { Preregistration as PreregistrationApi } from '@/modules/preregistration/api';
import VTooltip from 'v-tooltip';
import { createTestingPinia } from '@pinia/testing';
import filters from '@/filters';
import { initialize } from '../../mocks/google';
import { expect, vi } from 'vitest';
import { analytics } from '@/packages';
import { AnalyticsInstance } from 'analytics';
import { useModal } from '@/composables';
import {Preregistration} from "@/modules/processes/types";

const dialogSpy = vi.fn();

vi.mock('@/composables/useModal');
vi.mock('@/modules/preregistration/api');

const spyPush = vi.fn();
const spyBack = vi.fn();

vi.mock('vue-router/dist/vue-router.mjs', () => ({
  useRouter: () => ({
    push: spyPush,
    back: spyBack,
  }),
  useRoute: () => ({
    params: {
      id: '1',
    },
  }),
}));

vi.mock('@/packages');

const stageMock: PreRegistrationStage = {
  allowWaitingList: true,
  id: '1',
  process: {
    forceSuggestedGrade: true,
    gradeAgeRangeLink: null,
    id: '1',
    messageFooter:
      'Acompanhe a publicação do resultado classificatório e convocação no site da Prefeitura.',
    name: 'PROCESSO DE TESTES 2022',
    showPriorityProtocol: true,
    grades: [
      {
        endBirth: '2016-03-31',
        id: '22',
        name: '1º Ano',
        startBirth: '2015-03-31',
      },
    ],
    periods: [
      {
        id: '1',
        name: 'Matutino',
      },
    ],
    schoolYear: { year: 2022 },
    schools: [
      {
        id: '12345',
        latitude: 0,
        longitude: 0,
        name: 'ESCOLA MUNICIPAL DE ENSINO FUNDAMENTAL',
      },
    ],
    vacancies: [
      {
        available: 8,
        grade: 6,
        period: 1,
        school: 12345,
        total: 25,
      },
    ],
    fields: [
      {
        field: {
          group: 'RESPONSIBLE',
          id: '1',
          internal: 'responsible_name',
          name: 'Nome do(a) responsável',
          options: [],
          type: 'TEXT',
          description: null,
        },
        id: '262',
        order: 1,
        required: true,
        weight: 0,
      },
      {
        field: {
          group: 'STUDENT',
          id: '9',
          internal: 'student_name',
          name: 'Nome',
          options: [],
          type: 'TEXT',
          description: null,
        },
        id: '267',
        order: 2,
        required: true,
        weight: 0,
      },
    ],
    allowResponsibleSelectMapAddress: false,
    blockIncompatibleAgeGroup: false,
    onePerYear: false,
    waitingListLimit: 9,
    minimumAge: null,
  },
  radius: 1000,
  renewalAtSameSchool: false,
  status: 'OPEN',
  type: 'REGISTRATION',
  restrictionType: 'REGISTRATION_LAST_YEAR',
};

const onMounted = async (
  stage = stageMock,
  error?: boolean
) => {
  if (!error) {
    vi.mocked(PreregistrationApi.list).mockResolvedValueOnce({
      stage: stage,
      preregistration: undefined,
    });
  } else {
    vi.mocked(PreregistrationApi.list).mockRejectedValueOnce({
      stage: stage,
      preregistration: undefined,
    });
  }
  vi.mocked(analytics).mockImplementation(
    () =>
      ({
        page: vi.fn(),
      } as unknown as AnalyticsInstance)
  );

  vi.mocked(useModal).mockImplementation(() => ({
    dialog: dialogSpy,
  }));

  const wrapper = mount(PreRegistration, {
    global: {
      plugins: [createTestingPinia()],
      stubs: [
        'x-field',
        'google-maps',
        'google-maps-marker-component',
        'google-maps-markers',
      ],
      directives: {
        tooltip: VTooltip,
      },
      provide: {
        $filters: filters,
      },
      mocks: {
        google: vi.fn(() => google),
      },
    },
  });

  await wrapper.vm.$nextTick();

  await flushPromises();

  return wrapper;
};

describe('PreRegistration', () => {
  let wrapper: VueWrapper<InstanceType<typeof PreRegistration>>;

  beforeEach(async () => {
    initialize();

    wrapper = await onMounted();

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('the initial variables must be loaded with the correct data', () => {
    expect(wrapper.vm.stagePayload.stage).toMatchObject(stageMock);
    expect(wrapper.vm.fields.responsible.length).toBe(1);
    expect(wrapper.vm.fields.student.length).toBe(1);
    expect(wrapper.vm.responsible).toMatchObject({
      relationType: null,
      address: {
        postalCode: '',
        address: '',
        number: '',
        complement: '',
        neighborhood: '',
        lat: 0,
        lng: 0,
        city: '',
        cityIbgeCode: 0,
        stateAbbreviation: '',
      },
      useSecondAddress: false,
      secondAddress: {
        postalCode: '',
        address: '',
        number: '',
        complement: '',
        neighborhood: '',
        lat: 0,
        lng: 0,
        city: '',
        cityIbgeCode: 0,
        stateAbbreviation: '',
      },
      responsible_name: null,
    });
    expect(wrapper.vm.student).toMatchObject({
      grade: null,
      period: null,
      school: null,
      useSecondSchool: false,
      secondSchool: null,
      secondPeriod: null,
      match: null,
      student_name: null,
    });
    expect(wrapper.vm.newResponsible).toMatchObject({
      relationType: null,
      address: {
        postalCode: '',
        address: '',
        number: '',
        complement: '',
        neighborhood: '',
        lat: 0,
        lng: 0,
        city: '',
        cityIbgeCode: 0,
        stateAbbreviation: '',
      },
      useSecondAddress: false,
      secondAddress: {
        postalCode: '',
        address: '',
        number: '',
        complement: '',
        neighborhood: '',
        lat: 0,
        lng: 0,
        city: '',
        cityIbgeCode: 0,
        stateAbbreviation: '',
      },
      responsible_name: null,
    });
    expect(wrapper.vm.newStudent).toMatchObject({
      grade: null,
      period: null,
      school: null,
      useSecondSchool: false,
      secondSchool: null,
      secondPeriod: null,
      match: null,
      student_name: null,
    });
    expect(wrapper.vm.step).toBe('MATCH');
  });

  // test("if status isn't open, router push must be called to /", async () => {
  //   wrapper = await onMounted({
  //     stage: {
  //       ...stageMock,
  //       status: 'CLOSED',
  //     },
  //   });
  //
  //   await wrapper.vm.$nextTick();
  //
  //   expect(spyPush).toBeCalled();
  //   expect(spyPush).toBeCalledWith('/');
  // });
  //
  // test('if initial requisition crached, step has to be LOADING indefinitely', async () => {
  //   wrapper = await onMounted({
  //     stage: stageMock,
  //   }, true);
  //
  //   await wrapper.vm.$nextTick();
  //
  //   expect(wrapper.vm.step).toBe('LOADING');
  // });
  //
  // describe('prev', () => {
  //   beforeEach(async () => {
  //     wrapper = await onMounted();
  //
  //     await wrapper.vm.$nextTick();
  //   });
  //
  //   afterEach(() => {
  //     vi.clearAllMocks();
  //   });
  //
  //   test('step === RESPONSIBLE, must call router.back', async () => {
  //     wrapper.vm.step = 'RESPONSIBLE';
  //
  //     wrapper.vm.prev();
  //
  //     await wrapper.vm.$nextTick();
  //
  //     expect(spyBack).toBeCalled();
  //     expect(spyBack).toBeCalledTimes(1);
  //   });
  //
  //   test('step === STUDENT, step must be mutate to RESPONSIBLE', async () => {
  //     wrapper.vm.step = 'STUDENT';
  //
  //     wrapper.vm.prev();
  //
  //     await wrapper.vm.$nextTick();
  //
  //     expect(wrapper.vm.step).toBe('RESPONSIBLE');
  //   });
  //
  //   test('step === REVIEW, step must be mutate to STUDENT', async () => {
  //     wrapper.vm.step = 'REVIEW';
  //
  //     wrapper.vm.prev();
  //
  //     await wrapper.vm.$nextTick();
  //
  //     expect(wrapper.vm.step).toBe('STUDENT');
  //   });
  // });
  //
  // describe('next', () => {
  //   beforeEach(async () => {
  //     wrapper = await onMounted();
  //
  //     await wrapper.vm.$nextTick();
  //   });
  //
  //   afterEach(() => {
  //     vi.clearAllMocks();
  //   });
  //
  //   test('step === MATCH, submitMatch function must be called', async () => {
  //     const spySubmitMatch = vi.spyOn(wrapper.vm.ctx, 'submitMatch');
  //
  //     vi.mocked(PreregistrationApi.postMatch).mockResolvedValueOnce({
  //       id: 1,
  //       grade: 'string',
  //       period: 'string',
  //       school: 'string',
  //       initials: 'string',
  //       dateOfBirth: 'string',
  //     });
  //
  //     wrapper.vm.step = 'MATCH';
  //
  //     await wrapper.vm.next();
  //
  //     await wrapper.vm.$nextTick();
  //
  //     expect(spySubmitMatch).toBeCalled();
  //     expect(spySubmitMatch).toBeCalledTimes(1);
  //   });
  //
  //   test('step === RESPONSIBLE, step must be mutate to STUDENT', async () => {
  //     const spy = vi.spyOn(wrapper.vm.ctx, 'saveOnStorage');
  //
  //     wrapper.vm.step = 'RESPONSIBLE';
  //
  //     await wrapper.vm.next({
  //       address: {
  //         lat: 1,
  //         lng: 1,
  //       },
  //     });
  //
  //     await wrapper.vm.$nextTick();
  //
  //     expect(wrapper.vm.step).toBe('STUDENT');
  //     expect(spy).toBeCalled();
  //     expect(spy).toBeCalledTimes(1);
  //   });
  //
  //   test('step === RESPONSIBLE, but lat and lng is undefined step must continue to RESPONSIBLE', async () => {
  //     wrapper.vm.step = 'RESPONSIBLE';
  //
  //     await wrapper.vm.next({
  //       address: {},
  //     });
  //
  //     await wrapper.vm.$nextTick();
  //
  //     expect(wrapper.vm.step).toBe('RESPONSIBLE');
  //   });
  //
  //   test('step === STUDENT, submitToReview function must be called', async () => {
  //     const spySumitToReview = vi.spyOn(wrapper.vm.ctx, 'submitToReview');
  //     const spySaveOnStorage = vi.spyOn(wrapper.vm.ctx, 'saveOnStorage');
  //
  //     wrapper.vm.step = 'STUDENT';
  //
  //     await wrapper.vm.next();
  //
  //     expect(spySaveOnStorage).toBeCalledTimes(1);
  //     expect(spySumitToReview).toBeCalledTimes(1);
  //   });
  //
  //   test('step === REVIEW, submit function must be called', async () => {
  //     const spy = vi.spyOn(wrapper.vm.ctx, 'submit');
  //
  //     vi.mocked(PreregistrationApi.post).mockResolvedValueOnce({
  //       preregistrations: [
  //         {
  //           id: 1,
  //           protocol: 'string',
  //           code: 'string',
  //           type: 'string',
  //           date: '2022-06-02T00:00:00.000Z',
  //           position: 1,
  //           school: {
  //             id: 1,
  //             name: 'string',
  //             area_code: 'string',
  //             phone: 'string',
  //           },
  //           process: {
  //             showPriorityProtocol: true,
  //           },
  //         },
  //       ],
  //     });
  //
  //     wrapper.vm.step = 'REVIEW';
  //
  //     await wrapper.vm.next();
  //
  //     await flushPromises();
  //
  //     expect(spy).toBeCalledTimes(1);
  //   });
  // });
  //
  // describe('submitMatch', () => {
  //   beforeEach(async () => {
  //     wrapper = await onMounted();
  //
  //     await wrapper.vm.$nextTick();
  //   });
  //
  //   afterEach(() => {
  //     vi.clearAllMocks();
  //   });
  //
  //   test('submitMatch was be called successfully', async () => {
  //     const response = [
  //       {
  //         id: 1,
  //         initials: 'string',
  //         dateOfBirth: 'string',
  //         registration: {
  //           school: {
  //             id: 'string',
  //             name: 'string',
  //           },
  //           grade: {
  //             id: 'string',
  //             name: 'string',
  //           },
  //         },
  //       },
  //     ];
  //
  //     vi.mocked(PreregistrationApi.postMatch).mockResolvedValueOnce(response);
  //
  //     await wrapper.vm.submitMatch();
  //
  //     await wrapper.vm.$nextTick();
  //
  //     await flushPromises();
  //
  //     expect(wrapper.vm.student.match).toMatchObject(response[0]);
  //   });
  //
  //   test('submitMatch was be called but has no data', async () => {
  //     vi.mocked(PreregistrationApi.postMatch).mockResolvedValueOnce([]);
  //
  //     await wrapper.vm.submitMatch();
  //
  //     await wrapper.vm.$nextTick();
  //
  //     await flushPromises();
  //
  //     expect(wrapper.vm.student.match).toBe(undefined);
  //     expect(dialogSpy).toBeCalledWith({
  //       title: 'Aluno(a) não possui matrícula no ano passado',
  //       titleClass: 'danger',
  //       description: `Como o aluno não possui uma matrícula no ano passado não será possível seguir nesta opção. Clique em \'Voltar\' e efetue a inscrição em outro tipo de período.`,
  //     });
  //   });
  // });
  //
  // describe('submit', () => {
  //   beforeEach(async () => {
  //     wrapper = await onMounted();
  //
  //     await wrapper.vm.$nextTick();
  //   });
  //
  //   afterEach(() => {
  //     vi.clearAllMocks();
  //   });
  //
  //   test('submit was be called successfully', async () => {
  //     const response = {
  //       preregistrations: [],
  //     };
  //
  //     vi.mocked(PreregistrationApi.post).mockResolvedValueOnce(response);
  //
  //     await wrapper.vm.submit();
  //
  //     await wrapper.vm.$nextTick();
  //
  //     await flushPromises();
  //
  //     expect(wrapper.vm.preregistrations).toMatchObject(
  //       response.preregistrations
  //     );
  //     expect(wrapper.vm.step).toMatchObject('PROTOCOL');
  //   });
  //
  //   test('submit was called with error', async () => {
  //     const err = {
  //       response: {
  //         data: {
  //           errors: [
  //             {
  //               message: 'Erro aconteceu',
  //               extensions: {
  //                 message: 'Erro aconteceu denovo',
  //               },
  //             },
  //           ],
  //           preregistrations: [],
  //         },
  //       },
  //     };
  //
  //     vi.mocked(PreregistrationApi.post).mockRejectedValue(err);
  //
  //     await wrapper.vm.submit();
  //
  //     await wrapper.vm.$nextTick();
  //
  //     await flushPromises();
  //
  //     expect(dialogSpy).toBeCalledWith({
  //       title: err.response.data.errors[0].message,
  //       description: err.response.data.errors[0].extensions.message,
  //       titleClass: 'danger',
  //       iconLeft: 'status-red',
  //     });
  //   });
  // });
  //
  // describe('submitToReview', () => {
  //   beforeEach(async () => {
  //     wrapper = await onMounted();
  //
  //     await wrapper.vm.$nextTick();
  //   });
  //
  //   afterEach(() => {
  //     vi.clearAllMocks();
  //   });
  //
  //   test('submitToReview was be called successfully', async () => {
  //     wrapper.vm.step = 'STUDENT';
  //
  //     wrapper.vm.responsible = {
  //       responsible_cpf: '000.000.000-00',
  //       relationType: 'MOTHER',
  //     };
  //
  //     wrapper.vm.student = {
  //       student_cpf: '000.000.000-01',
  //       school: 1,
  //     };
  //
  //     wrapper.vm.submitToReview();
  //
  //     await wrapper.vm.$nextTick();
  //
  //     expect(wrapper.vm.step).toBe('REVIEW');
  //   });
  //
  //   test('submitToReview was be called with conditional errors', async () => {
  //     wrapper.vm.step = 'STUDENT';
  //
  //     wrapper.vm.responsible = {
  //       responsible_cpf: '000.000.000-00',
  //       relationType: 'MOTHER',
  //       address: {
  //         lat: 0,
  //         lng: 0,
  //       },
  //     };
  //
  //     wrapper.vm.student = {
  //       student_cpf: '000.000.000-00',
  //     };
  //
  //     wrapper.vm.submitToReview();
  //
  //     await wrapper.vm.$nextTick();
  //
  //     expect(wrapper.vm.step).toBe('STUDENT');
  //   });
  // });
});
