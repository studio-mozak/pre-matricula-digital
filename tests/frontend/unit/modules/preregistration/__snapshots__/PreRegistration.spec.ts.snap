// Vitest Snapshot v1

exports[`PreRegistration > should match snapshot 1`] = `
<div
  class="mt-5"
>
  <!--v-if-->
  <div>
    <form
      autocomplete="off"
      class="row"
    >
      
      
      
      <div
        class="col-12"
      >
        <div
          class="container-pmd"
        >
          <h2
            class="font-muli-20-primary mb-4"
          >
             Identifique o(a) aluno(a) para continuar com a inscrição: 
          </h2>
          <p>
             Preencha os dados abaixo para identificar o(a) 
            <strong>
              aluno(a)
            </strong>
             que será matrículado(a). 
          </p>
          <div
            class="row mt-5"
          >
            <div
              class="col-12 col-md-6"
            >
              <div
                class="autocomplete-field-wrapper"
                data-v-2c3a0553=""
                errors="false"
              >
                <label
                  class="autocomplete-label"
                  data-v-2c3a0553=""
                  for="student_name"
                >
                  Nome do(a) aluno(a)
                </label>
                <input
                  autocomplete="off"
                  class="autocomplete-input"
                  data-v-2c3a0553=""
                  id="student_name"
                  name="student_name"
                  placeholder=""
                  type="text"
                />
                <!--v-if-->
                <!--v-if-->
              </div>
            </div>
            <x-field-stub
              allowclear="false"
              bind="[object Object]"
              containerclass="col-12 col-md-6"
              disabled="false"
              errors="false"
              label="Data de nascimento do(a) aluno(a)"
              loading="false"
              name="student_date_of_birth"
              options=""
              rules="required|date|minimum_age:0"
              searchable="false"
              type="DATE"
            />
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
          </div>
        </div>
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      <div
        class="col-12"
      >
        <div
          class="row mt-5"
        >
          
          <div
            class="offset-md-3 col-md-3"
          >
            <button
              class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-gray-500 border-gray-500 x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100"
              role="button"
              style="font-size: 20px;"
              tabindex="0"
              type="button"
            >
              <span
                class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
              >
                <span
                  class="block"
                >
                  Voltar
                </span>
              </span>
              <transition-stub
                appear="false"
                css="true"
                name="x-transition--fade"
                persisted="false"
              >
                <!---->
              </transition-stub>
            </button>
          </div>
          <div
            class="col-md-3 mt-3 mt-md-0"
          >
            <button
              class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase btn btn-block btn-primary w-100 h-100 flex-row justify-content-center"
              disabled="false"
              role="button"
              style="font-size: 20px;"
              tabindex="0"
              type="button"
            >
              <span
                class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
              >
                <span
                  class="block"
                >
                  Prosseguir
                </span>
              </span>
              <transition-stub
                appear="false"
                css="true"
                name="x-transition--fade"
                persisted="false"
              >
                <!---->
              </transition-stub>
            </button>
          </div>
          
        </div>
      </div>
    </form>
  </div>
  <!--v-if-->
  <!--v-if-->
</div>
`;
